### PR TITLE
Add `Stream` usage example to README

### PR DESCRIPTION
**Pull request by hanlint**
Adds example usage of `Stream` feature.

### CODE REVIEW

**Review by bandish-shah — APPROVED** [review]
LGTM thanks!

File: README.md [hunk 1]
```diff
@@ -188,6 +188,25 @@ dataset = StreamingInsideWebVid(local=local, remote=remote, shuffle=True)
 
 ---
 
+## Seamless data mixing
+
+Easily experiment with dataset mixtures with `Stream`. Dataset sampling can be controlled in relative (proportion) or absolute (repeat or samples terms). During streaming, the different datasets are streamed, shuffled, and mixed seamlessly just-in-time.
```
Comment (bandish-shah) [inline README.md:193]: [nit] 
```suggestion
Easily experiment with dataset mixtures with [`Stream`](https://docs.mosaicml.com/projects/streaming/en/latest/api_reference/generated/streaming.Stream.html#stream). Dataset sampling can be controlled in relative (proportion) or absolute (repeat or samples terms). During training, the different datasets are streamed, shuffled, and mixed seamlessly just-in-time. 
```

Comment (karan6181) [reply README.md]: Fixed in https://github.com/mosaicml/streaming/pull/266/commits/3c566cc6de4b159665e570970c13d5d2f4acad95.

**Comment by knighton** [comment]
Um, guys? This code is broken.

**Comment by knighton** [comment]
```
streams = [
    Stream(remote='s3://datasets/c4', proportion=0.4),
    Stream(remote='s3://datasets/github', proportion=0.1),
    Stream(remote='gcs://datasets/my_internal', proportion=0.5),
]
dataset = StreamingDataset(
    streams=streams,
    samples_per_epoch=1e8,
)
```

**Pull request merged**